pkg/ipam/allocator/podcidr/podcidr.go

```diff
@@ -666,8 +666,16 @@ func (n *NodesPodCIDRManager) reuseIPNets(nodeName string, v4CIDR, v6CIDR []*net
 	// canAllocateIPv4PodCIDRs is set to true. It's fine that we don't allocate
 	// it now since this node will be put into the map of nodes that require
 	// to be allocated in the future.
-	if canAllocateIPv4PodCIDRs && len(n.v4CIDRAllocators) != 0 && len(v4CIDR) != 0 {
-		revertFunc, err = allocateIPNet(v4AllocatorType, n.v4CIDRAllocators, v4CIDR)
+	if canAllocateIPv4PodCIDRs && len(n.v4CIDRAllocators) != 0 {
+		if len(v4CIDR) != 0 {
+			revertFunc, err = allocateIPNet(v4AllocatorType, n.v4CIDRAllocators, v4CIDR)
+		} else {
+			// If the node does not have an IP address assigned to it, we need
+			// to allocate it because we have allocators available.
+			var newv4CIDR *net.IPNet
```

Comment:
Nit: is this needed as a standalone variable declaration?

Reply:
I think this patch clarifies the scope of the vars. Feel free to add or not.

```
diff --git a/pkg/ipam/allocator/podcidr/podcidr.go b/pkg/ipam/allocator/podcidr/podcidr.go
index 5f687b569..e6a07b273 100644
--- a/pkg/ipam/allocator/podcidr/podcidr.go
+++ b/pkg/ipam/allocator/podcidr/podcidr.go
@@ -649,8 +649,8 @@ func (n *NodesPodCIDRManager) reuseIPNets(nodeName string, v4CIDR, v6CIDR []*net
 	}
 
 	var (
-		revertStack revert.RevertStack
 		revertFunc  revert.RevertFunc
+		revertStack revert.RevertStack
 	)
 
 	defer func() {
@@ -667,12 +667,16 @@ func (n *NodesPodCIDRManager) reuseIPNets(nodeName string, v4CIDR, v6CIDR []*net
 	// it now since this node will be put into the map of nodes that require
 	// to be allocated in the future.
 	if canAllocateIPv4PodCIDRs && len(n.v4CIDRAllocators) != 0 {
+		var (
+			newv4CIDR *net.IPNet
+			err       error
+		)
+
 		if len(v4CIDR) != 0 {
 			revertFunc, err = allocateIPNet(v4AllocatorType, n.v4CIDRAllocators, v4CIDR)
 		} else {
 			// If the node does not have an IP address assigned to it, we need
 			// to allocate it because we have allocators available.
-			var newv4CIDR *net.IPNet
 			revertFunc, newv4CIDR, err = allocateFirstFreeCIDR(n.v4CIDRAllocators)
 			v4CIDR = append(v4CIDR, newv4CIDR)
 		}
@@ -691,12 +695,16 @@ func (n *NodesPodCIDRManager) reuseIPNets(nodeName string, v4CIDR, v6CIDR []*net
 	// it now since this node will be put into the map of nodes that require
 	// to be allocated in the future.
 	if canAllocateIPv6PodCIDRs && len(n.v6CIDRAllocators) != 0 {
+		var (
+			newv6CIDR *net.IPNet
+			err       error
+		)
+
 		if len(v6CIDR) != 0 {
 			revertFunc, err = allocateIPNet(v6AllocatorType, n.v6CIDRAllocators, v6CIDR)
 		} else {
 			// If the node does not have an IP address assigned to it, we need
 			// to allocate it because we have allocators available.
-			var newv6CIDR *net.IPNet
 			revertFunc, newv6CIDR, err = allocateFirstFreeCIDR(n.v6CIDRAllocators)
 			v6CIDR = append(v6CIDR, newv6CIDR)
 		}
```

Reply:
@christarazi yes because the last value set in err is supposed to be returned